libraries/AP_BLHeli/AP_BLHeli.cpp

```diff
@@ -1498,15 +1505,15 @@ void AP_BLHeli::log_bidir_telemetry(void)
 
     // ask the next ESC for telemetry
     uint8_t idx_pos = last_telem_esc;
-    for (uint8_t idx = (idx_pos + 1) % num_motors;
-        idx != idx_pos;
-        idx = (idx_pos + 1) % num_motors) {
-        uint16_t mask = 1U << motor_map[idx];
-        if (SRV_Channels::have_digital_outputs(mask)) {
-            last_telem_esc = idx;
+    uint8_t idx = (idx_pos + 1) % num_motors;
+    for (; idx != idx_pos; idx = (idx_pos + 1) % num_motors) {
```

Comment:
it's funny the idx assingment isn't in the for loop one line below..

Reply:
Yeah it was before, but that then missed the case of 1 motor which meant the loop was never entered

Reply:
I spent ages trying to figure out a good way of structuring this once I knew what the bug was - this was the best I could come up with